src/box/iproto.cc

```diff
@@ -1811,23 +1865,20 @@ iproto_msg_prepare(struct iproto_msg *msg, const char **pos, const char *reqend)
 		 type == IPROTO_ROLLBACK);
 
 	if (stream_id != 0 && request_is_not_for_stream) {
+		assert(!con->is_in_replication);
```

Comment:
Why did you add these assertions? Why did you move the check for replication request above? This assertion can actually fail if someone tries to set `stream_id` for a replication request.

Reply:
I had a misunderstanding, thank you.